Licensing: update W0019, 20, 21 (#8918)
DevExpress/devextreme-documentation: api-reference/50 Common/utils/zz Errors and Warnings/W0021.md

---
---
##### shortDescription
Appears when the DevExtreme runtime key verification fails.

---

> To resolve this warning, run the `devextreme-license` CLI tool to regenerate the DevExtreme runtime key and specify the newly generated key in [GlobalConfig](/api-reference/50%20Common/Object%20Structures/GlobalConfig '/Documentation/ApiReference/Common/Object_Structures/GlobalConfig/'). If the warning persists, ensure your DevExpress license key is registered correctly on your machine, then run `devextreme-license` again.

Reviewer: ```suggestion
To resolve this warning, run the `devextreme-license` CLI tool to regenerate the DevExtreme runtime key and specify the newly generated key in [GlobalConfig](/api-reference/50%20Common/Object%20Structures/GlobalConfig '/Documentation/ApiReference/Common/Object_Structures/GlobalConfig/'). If the warning persists, ensure that your DevExpress license key is registered correctly on your machine, then run `devextreme-license` again.
```

> If you have a valid DevExpress license key and this problem persists, submit a support ticket via the [DevExpress Support Center](https://supportcenter.devexpress.com/ticket/create).

Reviewer: ```suggestion
If you have a valid DevExpress license key and this problem persists, submit a support ticket in the [DevExpress Support Center](https://supportcenter.devexpress.com/ticket/create).
```

> For more information, refer to: [License Key for DevExpress Products](https://docs.devexpress.com/GeneralInformation/405494/trial-register/set-up-your-dev-express-license-key#manual-registration-additional-step-for-devextreme-jsts-developers)

Reviewer: ```suggestion
For additional information, refer to the following help topic: [License Key for DevExpress Products](https://docs.devexpress.com/GeneralInformation/405494/trial-register/set-up-your-dev-express-license-key#manual-registration-additional-step-for-devextreme-jsts-developers).
```

Appears when the license key verification fails.

---

> To resolve this warning, run the `devextreme-license` CLI tool to regenerate the DevExtreme runtime key and specify the newly generated key in [GlobalConfig](/api-reference/50%20Common/Object%20Structures/GlobalConfig '/Documentation/ApiReference/Common/Object_Structures/GlobalConfig/'). If the warning persists, ensure your DevExpress license key is registered correctly on your machine, then run `devextreme-license` again.

Reviewer 2: Let's add 'Appears when ...' e.g. Runtime license key is missing in the config or it is invalid.

it's better to add here the link to the general topic as well (at the end) https://docs.devexpress.com/GeneralInformation/405494/trial-register/set-up-your-dev-express-license-key#manual-registration-additional-step-for-devextreme-jsts-developers
